### PR TITLE
feat: updated project structure and added basic test cases to project

### PR DESCRIPTION
**Pull request by k-avy**
### feat: updated the project structure and added test case

* updated project structure to avoid import cycle
  *  added a different package for database where:
     * there is function related to migration of database
     * and populating database

* added basic test cases:
  * added API test cases for create, update, get license by shortname and search in licenses
  * added user test cases for create and get user by id

* updated meta to pagination meta for better discription

### CODE REVIEW

**Comment by sjha2048** [comment]
There can be some more changes in the project structure, let's go over it in our next call.
I will update this comment after our discussion.

File: pkg/api/api.go [hunk 1]
```diff
@@ -8,12 +8,36 @@ import (
...
+	// r is a default instance of gin engine
+	r := gin.Default()
+
+	// return error for invalid routes
+	r.NoRoute(HandleInvalidUrl)
+
+	// authorization not required for these routes
+	r.GET("/api/licenses/:shortname", GetLicense)
+	r.POST("/api/search", SearchInLicense)
+	r.GET("/api/licenses", FilterLicense)
```
Comment (avinal) [inline pkg/api/api.go:27]: Create an issue to merge these two endpoints into a single one.

File: pkg/models/types.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ type LicenseJson struct {
 // license retrieval operation.
 // It contains information that provides context and supplementary details
 // about the retrieved license data.
-type Meta struct {
+type PaginationMeta struct {
```
Comment (avinal) [inline pkg/models/types.go:59]: Create an issue to make this struct as free form key-value mapping field

File: pkg/models/types.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ type LicenseInput struct {
 	Shortname       string `json:"rf_shortname" binding:"required"`
 	Fullname        string `json:"rf_fullname" binding:"required"`
 	Text            string `json:"rf_text" binding:"required"`
-	Url             string `json:"rf_url" binding:"required"`
+	Url             string `json:"rf_url" `
```
Comment (avinal) [inline pkg/models/types.go:92]:
```suggestion
	Url             string `json:"rf_url"`
```

**Review by avinal — CHANGES_REQUESTED** [review]
Add the test command to the Github Actions as well.

**Pull request merged**